src/sisl/io/siesta/vibra.py

```diff
@@ -0,0 +1,257 @@
+        if self._state == -1:
+            self._sizes = self._r_sizes()
+            self._state = 0
+
+        if close:
+            self._close()
+
+    def _r_sizes(self):
+        """ Determine the dimension of the data stored in the vectors file
+        """
+        pos = self.fh.tell()
+        # Skip past header
+        self.readline() # Empty line
+        self.readline() # K point
+        self.readline() # Eigenmode index
+        self.readline() # Frequency
+        self.readline() # Eigenmode header
+        
+        # Determine number of atoms (i.e. rows per mode)
+        natoms = 0
+        while True:
+            line = self.readline()
+            if line.startswith('Eigenmode'):
+                break
+            else:
+                natoms += 1
+        self._natoms = natoms
+
+        # Determine number of modes
+        nmodes = 1
+        while True:
+            line = self.readline()
+            if line.startswith('k            ='):
+                break 
+            elif line.startswith('Eigenvector'):
+                nmodes += 1
+        self._nmodes = nmodes
+
+        # Rewind
+        self.seek(pos)
+
+    def _r_next_eigenmode(self):
+        """Reads the next phonon eigenmode in the vectors file.
+
+        Parameters
+        ----------
+        ik: integer
+            the (python) k index of the next eigenstate.
+
+        Returns
+        --------
+        EigenmodePhonon:
+            The next eigenmode.
+        """
+        # Skip empty line at the head of the file
+        self.readline()
+            
+        k = list(map(float, self.readline().split()[2:]))
+        if len(k) == 0:
+            raise GeneratorExit
+        # Read first eigenvector index 
+        
+        # Determine number of atoms (i.e. rows per mode)
+        state = np.empty((self._nmodes, self._natoms, 3), dtype=np.complex128)
+        c = np.empty(self._nmodes, dtype=np.float64)
+        for imode in range(self._nmodes):
+            line = self.readline()
+            if line.strip() == '':
+                # End of block for current k is indicated by empty line
+                break
+            
+            # Read frequency
+            c[imode] = float(self.readline().split('=')[1].strip()) * _hc_eV
+            
+            # Read real part of eigenmode
+            # Skip eigenmode header
+            self.readline()
+            for iatom in range(self._natoms):
+                line = self.readline().strip().lower()
+                state[imode,iatom,:].real = list(map(float, line.split()))
+
+            # Read imaginary part of eigenmode
+            # Skip eigenmode header
+            self.readline()
+            for iatom in range(self._natoms):
+                line = self.readline().strip().lower()
+                state[imode,iatom,:].imag = list(map(float, line.split()))
+
+        info = dict(k=self._convert_k(k))
```

Comment:
here `parent` would be nice, or you need a new internal method which creates the default info dict. I.e. in case where `parent` isn't defined, but `geometry` or `lattice` is

Reply:
I added parent. 

`parent` should always be defined unless neither `parent`, `geometry`, nor `lattice` are passed, because of:
```python
      def _setup(self, *args, **kwarg):
        # ...
        # ...   
        if parent is None:
            parent = geometry
        if parent is None:
            parent = lattice
 ```